### DISCUSSION TITLE
strange behaviour  !!

### DISCUSSION THREAD

**Discussion by ccdric** [zeux/pugixml#425] [Q&A] [answered]
Hello
probably I missed something obvious. 
In the next code (using GoogleTest Lib) the first test  failed but the second one succeeded strangely

```
    string expected("<?xml version=\"1.0\"?>\n<anomalies>\n	<erreur>toto</erreur>\n\
	<erreur>tititi</erreur>\n</anomalies>\n");
    pugi::xml_document verifCr;
    pugi::xml_parse_result result = verifCr.load_file(getFullPathCRAno(REAL_PATH).c_str());
    EXPECT_EQ(result,pugi::status_ok); // (1) Error Expected equality of these values: result Which is: 1  pugi::status_ok  Which is: 0


    ostringstream contenue;
    verifCr.save(contenue);
    EXPECT_EQ(contenue.str(), expected); // (2) OK
```

if I change something in var "expected" (for ex : replace first node "anomalies" by node "dummy"), it finds the difference, in the file although it was not supposed to read.
And now the  second test failed also !!
```
/..../myProj/utest/src/T18_EnvoisCRDoit.cpp:248: Failure
Expected equality of these values:
  result
    Which is: 1
  pugi::status_ok
    Which is: 0
/..../myProj/utest/src/T18_EnvoisCRDoit.cpp:248: Failure
Expected equality of these values:
  contenue.str()
    Which is: "<?xml version=\"1.0\"?>\n<anomalies>\n\t<erreur>toto</erreur>\n\t<erreur>titi</erreur>\n</anomalies>\n"
  attendu
    Which is: "<?xml version=\"1.0\"?>\n<dumy>\n\t<erreur>toto</erreur>\n\t<erreur>tititi</erreur>\n</anomalies>\n"
With diff:
@@ -1,5 +1,5 @@
 <?xml version=\"1.0\"?>
-<anomalies>
+<dumy>
 \t<erreur>toto</erreur>
-\t<erreur>titi</erreur>
+\t<erreur>tititi</erreur>
 </anomalies>\n
```
Can you help me and explain how pugi lib  can in the same time say "file not found" (return 1) and really read the content in this "file not found"

**Comment by ccdric** [comment]
sorry please read "expected" in place of "attendu" in the second code quoted part of my message
(I replaced only this var name for make it easily comprehensive  for English native people)

**Comment by zeux** [comment, marked as answer]
pugi::xml_parse_result is a struct. You're looking for either

    EXPECT_EQ(result.status, pugi::status_ok);

or

    EXPECT(result);

What you see is the implicit conversion of result object to a boolean, and a subsequent attempt to compare that boolean (true) to status_ok.

**Reply by ccdric** [reply]
thank you 
it's right : 
```
    EXPECT_TRUE(result);                            // OK
    EXPECT_EQ(result.status, pugi::status_ok);     // OK

    ostringstream contenue;
    verifCr.save(contenue);
    EXPECT_EQ(contenue.str(), attendu);             // OK
```
